File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> um;
        for(int el:nums){
            if(um[el] == 1){
                return true;
            }else{
                um[el]++;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        majority = n / 2

        my_map = dict()

        for num in nums:
            my_map[num] = my_map.get(num, 0) + 1

        for key in my_map.keys():
            if my_map.get(key) >= majority:
                return key
